design/aws-ecs-service-discovery-integration.md

```diff
@@ -0,0 +1,235 @@
+# AWS ECS - Support for AWS Cloud Map (Service Discovery) Integration
+
+To address issue [#1554](https://github.com/awslabs/aws-cdk/issues/1554), the
+ECS construct library should provide a way to set
+[`serviceRegistries`](https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_CreateService.html#ECS-CreateService-request-serviceRegistries)
+on their ECS service at the L1 level.
+
+
+## General approach
+Rather than having the customer instantiate Cloud Map constructs directly and
+thus be forced to learn the API of a separate construct library, we should
```

Comment:
I think this is probably fine, but I would also expect to be able to pass in an `INamespace` if I know what I'm doing (same as we do for passing role).

Reply:
Honestly, I wonder if this is not all a whole load of design pain for not a lot of gain.

If we make `ecs.Service` take a `cloudmap.IService`, are we not done?

Reply:
Sure, you could do that but that would require having the customer have to know about a larger set of properties to create the service, and then also forcing them to learn which what fields from that resource need to be passed into the service as a ServiceRegistry. 

As for passing in an entire `INamespace`, that is perhaps possible (and something I need to think about for public namespaces, since the ECS console doesn't create one for you the way it does a Private DNS namespace) but I was coming at it from the point of view that the only thing the customer cares about is a domain name (this was also validated by ECS engineers involved with cloudmap integration/service mesh).